File: frontend/src/types/processo.ts
export type FaseProcessual = "conhecimento" | "recursal" | "execucao";
export type StatusProcesso = "ativo" | "suspenso" | "arquivado" | "encerrado";
export type OrigemMovimentacao = "datajud" | "escavador" | "manual";

export interface ClienteResumo {
  id: string;
  nome_razao_social: string;
}

export interface Processo {
  id: string;
  numero_cnj: string;
  vara: string | null;
  tribunal: string | null;
  comarca: string | null;
  tipo_acao: string | null;
  fase_processual: FaseProcessual;
  polo_ativo: string | null;
  polo_passivo: string | null;
  status: StatusProcesso;
  data_distribuicao: string | null;
  valor_causa: string | null;
  cliente_id: string;
  advogado_responsavel_id: string | null;
  ultima_consulta_datajud: string | null;
  cliente?: ClienteResumo | null;
}

export interface ProcessoListResponse {
  items: Processo[];
  total: number;
  page: number;
  page_size: number;
}

export interface ProcessoCreateInput {
  numero_cnj: string;
  cliente_id: string;
  vara?: string;
  tribunal?: string;
  comarca?: string;
  tipo_acao?: string;
  fase_processual?: FaseProcessual;
  polo_ativo?: string;
  polo_passivo?: string;
  data_distribuicao?: string;
  valor_causa?: string;
}

export interface ProcessoUpdateInput {
  vara?: string;
  tribunal?: string;
  comarca?: string;
  tipo_acao?: string;
  fase_processual?: FaseProcessual;
  polo_ativo?: string;
  polo_passivo?: string;
  status?: StatusProcesso;
  data_distribuicao?: string;
  valor_causa?: string;
}

export interface Movimentacao {
  id: string;
  processo_id: string;
  data_movimentacao: string;
  descricao: string;
  origem: OrigemMovimentacao;
}

export interface MovimentacaoCreateInput {
  data_movimentacao: string;
  descricao: string;
}

File: backend/app/schemas/processo.py
import uuid
from datetime import date, datetime
from decimal import Decimal

from pydantic import BaseModel, ConfigDict, field_validator

from app.core.validators import numero_cnj_valido, somente_digitos
from app.models.enums import FaseProcessual, OrigemMovimentacao, StatusProcesso


class ProcessoBase(BaseModel):
    numero_cnj: str
    vara: str | None = None
    tribunal: str | None = None
    comarca: str | None = None
    tipo_acao: str | None = None
    fase_processual: FaseProcessual = FaseProcessual.CONHECIMENTO
    polo_ativo: str | None = None
    polo_passivo: str | None = None
    status: StatusProcesso = StatusProcesso.ATIVO
    data_distribuicao: date | None = None
    valor_causa: Decimal | None = None

    @field_validator("numero_cnj")
    @classmethod
    def valida_numero_cnj(cls, v: str) -> str:
        if not numero_cnj_valido(v):
            raise ValueError("Número CNJ inválido")
        return somente_digitos(v)


class ProcessoCreate(ProcessoBase):
    cliente_id: uuid.UUID
    advogado_responsavel_id: uuid.UUID | None = None


class ProcessoUpdate(BaseModel):
    vara: str | None = None
    tribunal: str | None = None
    comarca: str | None = None
    tipo_acao: str | None = None
    fase_processual: FaseProcessual | None = None
    polo_ativo: str | None = None
    polo_passivo: str | None = None
    status: StatusProcesso | None = None
    data_distribuicao: date | None = None
    valor_causa: Decimal | None = None
    advogado_responsavel_id: uuid.UUID | None = None


class ClienteResumo(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    nome_razao_social: str


class ProcessoRead(ProcessoBase):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    cliente_id: uuid.UUID
    advogado_responsavel_id: uuid.UUID | None
    ultima_consulta_datajud: datetime | None
    cliente: ClienteResumo | None = None


class ProcessoListResponse(BaseModel):
    items: list[ProcessoRead]
    total: int
    page: int
    page_size: int


class MovimentacaoCreate(BaseModel):
    data_movimentacao: datetime
    descricao: str


class MovimentacaoRead(MovimentacaoCreate):
    model_config = ConfigDict(from_attributes=True)

    id: uuid.UUID
    processo_id: uuid.UUID
    origem: OrigemMovimentacao
